molecule/scenario.py

```diff
@@ -240,16 +237,11 @@ def ephemeral_directory(path=None):
     """
     d = os.getenv('MOLECULE_EPHEMERAL_DIRECTORY')
     if not d:
-        # Darwin is the only known platform that returns user-isolated tempdirs
-        # so we do not need to make them unique to each user.
-        if platform.system() == 'Darwin':
-            d = tempfile.gettempdir()
-        else:
-            d = os.path.join(tempfile.gettempdir(), 'mol.' + getpass.getuser())
-    else:
-        d = os.path.abspath(d)
-    if path:
-        d = os.path.abspath(os.path.join(d, path))
-        if not os.path.isdir(d):
-            Path(d).mkdir(mode=0o700, parents=True, exist_ok=True)
+        d = os.getenv("XDG_CACHE_HOME", os.path.expanduser("~/.cache"))
+    d = os.path.abspath(os.path.join(d, path if path else 'molecule'))
+
+    if not os.path.isdir(d):
+        os.umask(0o077)
+        Path(d).mkdir(mode=0o700, parents=True, exist_ok=True)
```

Comment:
Given the original comment in https://github.com/ansible/molecule/pull/2089#discussion_r294474426, we're now OK with this `parents=True` stuff?

Reply:
It is ok for me. I think that it would make no sense to question what-if someone already hacked you machine scenario. 

I don't really care what happens if user is using a custom ephemeral directory which points to a non existing location or a location what has previously hacked.

Once the user starts to customize under the hood options it is its responsibility how he uses them.

Now the default behaviour is safe, much safer that what used to be.

Reply:
Yeah, sorry I was unclear about the precise reason that parents 9and exists_ok as well) are dangerous.  ```parents=True``` and ```exist_ok=True``` are especially bad in a world-writable directory because they mean that we are unaware whether we have created the directory structure or an attacker has created the directory structure.  When we root the directory structure in the user's home directory, we are okay to use them because we consider that the user's home directory is only writable by the user to begin with.  Applications like ssh, will go a step further and verify permissions on the directory structure to make sure that an attacker cannot write to any of the directories involved.

So this new code looks like it closes the hole.